api/v1alpha1/shared_types.go

```diff
@@ -333,3 +339,22 @@ type KubernetesHorizontalPodAutoscalerSpec struct {
 	// +optional
 	Behavior *autoscalingv2.HorizontalPodAutoscalerBehavior `json:"behavior,omitempty"`
 }
+
+// MergeType defines the type of merge operation
+type MergeType string
+
+const (
+	// StrategicMerge indicates a strategic merge patch type
+	StrategicMerge MergeType = "StrategicMerge"
+	// JSONMerge indicates a JSON merge patch type
+	JSONMerge MergeType = "JSONMerge"
+)
+
+// KubernetesPatchSpec defines how to perform the patch operation
+type KubernetesPatchSpec struct {
+	// Type is the type of merge operation to perform
+	Type MergeType `json:"type"`
+
+	// Object contains the raw configuration for merged object
+	Object apiextensionsv1.JSON `json:"object"`
```

Comment:
prefer if this field is called `value` which matches what is used in other APIs in the project
https://gateway.envoyproxy.io/latest/api/extension_types/#jsonpatchoperation
https://gateway.envoyproxy.io/latest/api/extension_types/#proxybootstrap